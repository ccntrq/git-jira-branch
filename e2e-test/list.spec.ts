import {beforeEach, describe, expect, it} from 'vitest';
import {
  type Directory,
  createBranch,
  runApp,
  setupTmpDir,
  switchBranch,
} from './util';

describe('git-jira-branch list', () => {
  let tmpDir: Directory;
  const listCommand = (...args: Array<string>) =>
    runApp(tmpDir, 'list', ...args);

  beforeEach(async () => {
    const [dir, cleanup] = await setupTmpDir();
    tmpDir = dir;
    return cleanup;
  });

  it('starts app and outputs help', async () => {
    const res = await listCommand('--help');
    expect(res).toMatch(/git-jira-branch/);
  });

  it('list subcommand prints associated branches with current', () => {
    createBranch(tmpDir, 'feat/GCJB-1-test-ticket-1');
    createBranch(tmpDir, 'feat/GCJB-2-test-ticket-2');
    createBranch(tmpDir, 'feat/GCJB-3-test-ticket-3-current');

    expect(listCommand()).resolves.toMatchSnapshot();
  });

  it('list subcommand prints associated branches without current', () => {
    createBranch(tmpDir, 'feat/GCJB-1-test-ticket-1');
    createBranch(tmpDir, 'feat/GCJB-2-test-ticket-2');
    createBranch(tmpDir, 'feat/GCJB-3-test-ticket-3');
    switchBranch(tmpDir, 'master');

    expect(listCommand()).resolves.toMatchSnapshot();
  });
});
