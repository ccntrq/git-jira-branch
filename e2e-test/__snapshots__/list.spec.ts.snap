// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`git-jira-branch list > list subcommand prints associated branches with current 1`] = `
"  feat/GCJB-1-test-ticket-1
  feat/GCJB-2-test-ticket-2
* [0;32mfeat/GCJB-3-test-ticket-3-current[0m
"
`;

exports[`git-jira-branch list > list subcommand prints associated branches without current 1`] = `
"feat/GCJB-1-test-ticket-1
feat/GCJB-2-test-ticket-2
feat/GCJB-3-test-ticket-3
"
`;
